### PR TITLE
feat: implementing changes in add edit per diem [5]

### PR DESCRIPTION
**Pull request by Harshal141**
### Description
implemented the new reports service to add edit per diem page which now uses platform API

## Clickup
https://app.clickup.com/t/86cuqardf

### CODE REVIEW

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 1]
```diff
@@ -662,6 +665,9 @@ export class AddEditPerDiemPage implements OnInit {
   }
 
   getEditExpense(): Observable<UnflattenedTransaction> {
+    console.log('-------------------');
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:668]: Remove console logs

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 2]
```diff
@@ -1066,9 +1072,20 @@ export class AddEditPerDiemPage implements OnInit {
...
     );
 
-    this.reports$ = this.reportService
-      .getFilteredPendingReports({ state: 'edit' })
-      .pipe(map((reports) => reports.map((report) => ({ label: report.rp.purpose, value: report }))));
+    this.reports$ = this.platformReportService
+      .getAllReportsByParams({ state: 'in.(DRAFT,APPROVER_PENDING,APPROVER_INQUIRY)' })
+      .pipe(
+        // Filter out partially approved reports
+        map((reports) =>
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:1079]: Combine these two maps in one

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 3]
```diff
@@ -2035,7 +2052,7 @@ export class AddEditPerDiemPage implements OnInit {
               map((savedEtxn) => savedEtxn && savedEtxn.tx),
               switchMap((tx) => {
                 const formValue = this.getFormValues();
-                const selectedReportId = formValue.report && formValue.report.rp && formValue.report.rp.id;
+                const selectedReportId = formValue.report && formValue.report.id;
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:2055]: `formValue.report?.id`

**Review by suyashpatil78 — APPROVED** [review]
Remove console logs and please to fix the comments

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts (hunk 2 above)
Comment (Harshal141) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:1079]: i can use this 
this.reports$ = this.platformReportService
  .getAllReportsByParams({ state: 'in.(DRAFT,APPROVER_PENDING,APPROVER_INQUIRY)' })
  .pipe(
    map((reports: Report[]) =>
      reports
        .filter((report) => !report.approvals.some((approval) => approval.state === 'APPROVAL_DONE'))
        .map((report) => ({ label: report.purpose, value: report }))
    )
  ) as Observable<{ label: string; value: Report }[]>;
but it still uses 2 maps can you give an idea how to do it

Comment (Julias0) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:1076]: I think Suyash meant that you combine the map pipes into one. The .map usage inside is completely fine can

**Review by Julias0 — APPROVED** [review]

**Pull request merged**